### PR TITLE
Let logic find the closest block of a certain type (units)

### PR DESCRIPTION
**Pull request by ThePythonGuy3**
This lets you do things like this one: https://discord.com/channels/391020510269669376/742769933926269069/768777006367113236

### CODE REVIEW

File: core/src/mindustry/logic/LExecutor.java [hunk 1]
```diff
@@ -260,6 +261,19 @@ public void run(LExecutor exec){
                             res = b == null ? null : b.tile;
                             build = true;
                         }
+                        case block -> {
+                            if(exec.obj(block) instanceof Block) {
+                                if (exec.bool(enemy)) {
+                                    Building b = Units.findEnemyTile(unit.team, unit.x, unit.y, Float.MAX_VALUE, e -> !e.dead() && e.block == exec.obj(block));
```
Comment (Anuken) [inline core/src/mindustry/logic/LExecutor.java:267]: This is very inefficient and not an acceptable solution. You can't scan the whole map every time you want to fetch a block.

File: core/src/mindustry/logic/LExecutor.java [hunk 2]
```diff
@@ -260,6 +261,19 @@ public void run(LExecutor exec){
...
+                                    res = b == null ? null : b.tile;
+                                    build = true;
+                                } else {
```
Comment (Anuken) [inline core/src/mindustry/logic/LExecutor.java:270]: [Read the style guide!](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#style-guidelines) I'm getting sick of saying this.

**Review by Anuken — CHANGES_REQUESTED** [review]

File: core/src/mindustry/logic/LExecutor.java (hunk 2 above)
Comment (ThePythonGuy3) [reply core/src/mindustry/logic/LExecutor.java]: im very sorry, fixed

File: core/src/mindustry/logic/LExecutor.java (hunk 1 above)
Comment (ThePythonGuy3) [reply core/src/mindustry/logic/LExecutor.java:267]: Then what should i do?

Comment (Anuken) [reply core/src/mindustry/logic/LExecutor.java:267]: Not sure what the best solution is. Could try a loop through all building entities, or a cache that tracks buildings by type.

Comment (ThePythonGuy3) [reply core/src/mindustry/logic/LExecutor.java:267]: hmmm ill see ok

**Comment by Anuken** [comment]
Caching blocks by type may be complex, so I'll implement this myself when I have time.

**Comment by ThePythonGuy3** [comment]
okay thank you

**Pull request closed without merging**